packages/api/cms-api/src/user-permissions/decorators/required-permission.decorator.ts

```diff
@@ -0,0 +1,29 @@
+import { EntityClass, EntityManager } from "@mikro-orm/core";
+import { CustomDecorator, SetMetadata } from "@nestjs/common";
+import { Request } from "express";
+
+import { CurrentUser } from "../dto/current-user";
+import { ContentScope } from "../interfaces/content-scope.interface";
+import { Permission } from "../interfaces/user-permission.interface";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type RequiredPermissionArgs<ArgsType = any> = {
+    args: ArgsType;
+    getScopeFromEntity: (entityClass: EntityClass<object>, id: string) => Promise<ContentScope>;
+    user: CurrentUser;
+    entityManager: EntityManager;
+    request: Request;
+};
+
+type RequiredPermissionOptions = {
+    skipScopeCheck?: boolean;
+};
+
+export type RequiredPermission = {
+    requiredPermission: (keyof Permission)[];
+    options: RequiredPermissionOptions | undefined;
+};
+
+export const RequiredPermission = (requiredPermission: (keyof Permission)[], options?: RequiredPermissionOptions): CustomDecorator<string> => {
```

Comment:
not related to this PR: do you think we could use this technique to define permissions also in admin? (where we had the discussion string vs enum)

Reply:
I'll create a PR with module augmentation for the permissions in the admin.